src/osd/ECBackend.cc

```diff
@@ -1537,62 +1538,65 @@ int ECBackend::objects_read_sync(
   return -EOPNOTSUPP;
 }
 
+static bool should_partial_read(
+  bool fast_read)
+{
+  // Don't partial read if we are doing a fast_read
+  if (fast_read) {
+    return false;
+  }
+  return true;
+}
+
 void ECBackend::objects_read_async(
   const hobject_t &hoid,
-  const list<pair<boost::tuple<uint64_t, uint64_t, uint32_t>,
-             pair<bufferlist*, Context*> > > &to_read,
+  const list<pair<ECCommon::ec_align_t,
+                  pair<bufferlist*, Context*>>> &to_read,
   Context *on_complete,
   bool fast_read)
 {
-  map<hobject_t,std::list<boost::tuple<uint64_t, uint64_t, uint32_t> > >
-    reads;
+  map<hobject_t,std::list<ec_align_t>> reads;
 
   uint32_t flags = 0;
   extent_set es;
-  for (list<pair<boost::tuple<uint64_t, uint64_t, uint32_t>,
-	 pair<bufferlist*, Context*> > >::const_iterator i =
-	 to_read.begin();
-       i != to_read.end();
-       ++i) {
-    pair<uint64_t, uint64_t> tmp =
-      sinfo.offset_len_to_stripe_bounds(
-	make_pair(i->first.get<0>(), i->first.get<1>()));
-
+  for (const auto& [read, ctx] : to_read) {
+    pair<uint64_t, uint64_t> tmp;
+    if (!cct->_conf->osd_ec_partial_reads || !should_partial_read(fast_read)) {
```

Comment:
I see different intentions behind the checks.

1. `osd_ec_partial_reads` is supposed to allow moving back to the old logic at run-time in case of a mistake found in the field. It should be shortly living. I perceive it as some kind of _insurance_.
2. `should_partial_read()` is part of the core logic; it will be still there after dropping the configurable.